app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardScreen.kt

```diff
@@ -130,17 +130,19 @@ fun KeyboardScreen(
     val backdropColor = MaterialTheme.colorScheme.background
     val backdropPadding = 6.dp
     val keyPadding = settings?.keyPadding ?: DEFAULT_KEY_PADDING
-    val legendSize = settings?.keySize ?: DEFAULT_KEY_SIZE
+    val legendHeight = settings?.keySize ?: DEFAULT_KEY_SIZE
+    val legendWidth = (if ((settings?.keyWidth ?: 0) > 0) settings?.keyWidth else legendHeight) ?: legendHeight
```

Comment:
You can tell that 0 is being used as a sub for null here, otherwise this could be as easy as

```kt
val legendWidth = settings?.keyWidth ?: legendHeight
```